main/src/library/List.flix

```diff
@@ -549,6 +549,17 @@ namespace List {
                 if (f(x)) x :: r else r
     }
 
+    ///
+    /// Returns the sublist of `xs` without the last element and, if it exists, the last element
+    ///
+    pub def init(xs: List[a]): (List[a], Option[a]) = initHelper(xs, Nil)
```

Comment:
I am not sure about this signature. I can understand where it comes from, but at the same time it is inconsistent with `init` functions in other programming languages. I will think a bit about what to do.

Reply:
In particular, why not return `Option[List[a]]` -- that would be compatible with other languages.

Reply:
If ignoring the last element of the list is alright simply returning `List[a]` should be fine, as long as you don't care to distinguish between the `Nil` and `1 :: Nil` cases in the signature.

Reply:
What does Scala do? In Haskell it is an error to call `init` on `Nil`.

Reply:
Scala throws `java.lang.UnsupportedOperationException: init of empty list`. And Haskell's design is, in my opinion, untenable because it breaks with the promise of the function signature.

Reply:
OK; so both languages consider it a mistake to call `init` on `Nil`. This means that `init` is really a partial function, and we usually implement such functions by returning `Option`. You proposal would be "inconsistent" with Scala and Haskell. Does it make sense? :)

Reply:
It makes perfect sense. I was just wondering out loud if following that decision is the best choice, or if swallowing the `Nil` case was acceptable. But since `List.init(NiL) |> Option.getWithDefault(Nil)` gets the same result (when `init` returns `Option[List[a]]`) it's probably a sensible choice to go with the mot explicit signature.

Reply:
Sure, I'll change the signature

Reply:
Your original version is easy to obtain anyway, just do a reverse and a pattern match :)